src/NuGet.Core/NuGet.Versioning/VersionRange.cs

```diff
@@ -402,5 +403,59 @@ public override int GetHashCode()
         {
             return base.GetHashCode();
         }
+
+        /// <summary>
+        /// Format the version range as a package version range specification string.
+        /// </summary>
+        /// <param name="packageId">The package id to be used in the dependency group.</param>
+        /// <example>
+        /// Package.Id >= 1.0.0
+        /// </example>
+        /// <remarks>
+        /// Used to label dependency groups or alternate packages.
+        /// </remarks>
+        public string ToPackageVersionRangeSpecificationString(string packageId)
```

Comment:
Could you add some tests please.

Reply:
This is existing code so I assumed we had tests. Will check.

Reply:
Reverted to use `PrettyPrint` instead :) https://github.com/NuGet/NuGet.Client/pull/2939/commits/03838cdb06c60782d11e46ed60e7266125924609